sdk/synapse/azure-synapse-accesscontrol/azure/synapse/accesscontrol/aio/operations/_role_assignments_operations.py

```diff
@@ -40,128 +50,101 @@ def __init__(self, client, config, serializer, deserializer) -> None:
         self._config = config
 
     async def check_principal_access(
-        self,
-        subject: "_models.SubjectInfo",
-        actions: List["_models.RequiredAction"],
-        scope: str,
-        **kwargs
+        self, *, subject: "_models.SubjectInfo", actions: List["_models.RequiredAction"], scope: str, **kwargs: Any
     ) -> "_models.CheckPrincipalAccessResponse":
         """Check if the given principalId has access to perform list of actions at a given scope.
 
-        :param subject: Subject details.
-        :type subject: ~azure.synapse.accesscontrol.models.SubjectInfo
-        :param actions: List of actions.
-        :type actions: list[~azure.synapse.accesscontrol.models.RequiredAction]
-        :param scope: Scope at which the check access is done.
-        :type scope: str
+        :keyword subject: Subject details.
```

Comment:
subject is not kwarg.

Reply:
I should've mentioned this in the email, but part of the "version-intolerant" design of autorest is returning a convenience layer, where everything _except_ path and body parameters are kwargs. So this is by design